server/app.ts

```diff
@@ -460,7 +462,7 @@ apiRouter.route("/data/tag/:tag").delete(authenticateWithReject, async (request,
 
 apiRouter.route("/search").get(authenticateWithReject, async (request, response) => {
 	let query: string = request.query.q || "";
-	let queryRegExp = new RegExp(query, "i");
+	let queryRegExp = new RegExp(reEscape(query), "i");
```

Comment:
I think it should because query is a string that can contain characters with special meaning for the regex parser